pkg/registry/core/service/allocator/storage/storage.go

```diff
@@ -62,7 +62,7 @@ var _ rangeallocation.RangeRegistry = &Etcd{}
 // NewEtcd returns an allocator that is backed by Etcd and can manage
 // persisting the snapshot state of allocation after each allocation is made.
 func NewEtcd(alloc allocator.Snapshottable, baseKey string, resource schema.GroupResource, config *storagebackend.Config) (*Etcd, error) {
-	storage, d, err := generic.NewRawStorage(config, nil)
+	storage, d, err := generic.NewRawStorage(config.ForResource(resource), nil)
```

Comment:
And given my comment above, probably stop taking `resource` as a param of `NewEtcd` function and taking it directly from the ConfigForResource in Line 79

Reply:
done